src/lib.rs

```diff
@@ -190,10 +194,53 @@ impl KmerCountTable {
     // Default sort by count
     // Option sort kmers lexicographically
 
-    // TODO: Add method "dump_hash"
-    // Output tab delimited hash:count pairs
-    // Default sort by count
-    // Option sort on keys
+    /// Dump hash:count pairs, sorted by count (default) or by hash key.
+    ///
+    /// # Arguments
+    /// * `file` - Optional file path to write the output. If not provided, returns a list of tuples.
+    /// * `sortkeys` - Optional flag to sort by hash keys (default: False).
+    ///
+    /// By default, the records are sorted by count in ascending order. If two records have the same
+    /// count value, they are sorted by the hash value. If `sortkeys` is set to `True`, sorting is done
+    /// by the hash key instead.
+    #[pyo3(signature = (file=None, sortkeys=false))]
```

Comment:
curious about needing to sort - for large k-mer collections, this is going to cost real memory. wouldn't a better default be to not sort, and then allow sorting?

Reply:
My thinking was that it would be useful to default sort on counts then hashes:
- Gives same dump file irrespective of the order that kmers were consumes (same data, same output)
- Easy to do a quick visual sanity check on the output with head / tail

Opt 1: Add a sortcounts opt (+ logic to deal with both sortkeys and sortcounts being set)
Opt 2: Keep default sortcounts, add opt nosort

Leaning toward Opt 1, can just use sortcounts in the example docs.

Reply:
if you're up for adding sortcounts option now, that'd be great - but can also be added later. (since I tend to work in bursts, I'll often punt things like this to an issue to be tackled later. goal is to not forget, but not necessarily pressure ourselves to do things quickly :))

Reply:
nosort is also fine, too!